tests/ops/op_math/test_exp.py

```diff
@@ -435,23 +439,33 @@ def test_decomposition_into_pauli_rot(self, base, base_string):
         pr = op.decomposition()[0]
         assert qml.equal(pr, qml.PauliRot(3.21, base_string, base.wires))
 
-    @pytest.mark.parametrize("op_class", has_unitary_generator_types)
-    def test_generator_decomposition(self, op_class):
+    @pytest.mark.parametrize("op_name", all_qubit_operators)
+    def test_generator_decomposition(self, op_name):
         """Check that Exp decomposes into a specific operator if ``base`` corresponds to the
         generator of that operator."""
+        op_class = getattr(qml.ops.qubit, op_name)
+
+        if not op_class.has_generator:
+            pytest.skip("Operator does not have a generator.")
+
         if op_class in {qml.DoubleExcitationMinus, qml.DoubleExcitationPlus}:
             pytest.skip("qml.equal doesn't work for `SparseHamiltonian` generators.")
 
         phi = 1.23
 
         wires = [0, 1, 2] if op_class.num_wires is AnyWires else list(range(op_class.num_wires))
 
+        # PauliRot and PCPhase each have an extra required arg
         if op_class is qml.PauliRot:
             op = op_class(phi, pauli_word="XYZ", wires=wires)
+        elif op_class is qml.PCPhase:
+            op = op_class(phi, dim=8, wires=wires)
```

Comment:
Can we test using `dim < 8` so that this operator is non-trivial?

Reply:
Ah of course - thanks for spotting!